src/WingetCreateCore/Common/PackageParser.cs

```diff
@@ -183,12 +183,14 @@ public static string GetFileHash(string path)
         /// <param name="paths">Paths to packages to extract metadata from.</param>
         /// <param name="detectedArchOfInstallers">List of DetectedArch objects that represent each installers detected architectures.</param>
         /// <param name="newInstallers">List of newly parsed installer nodes.</param>
+        /// <param name="installerArchOverrideMap">Dictionary that maps the installer URL with the override architecture.</param>
         public static void UpdateInstallerNodesAsync(
             InstallerManifest installerManifest,
             IEnumerable<string> installerUrls,
             IEnumerable<string> paths,
             out List<DetectedArch> detectedArchOfInstallers,
-            out List<Installer> newInstallers)
+            out List<Installer> newInstallers,
+            Dictionary<string, InstallerArchitecture> installerArchOverrideMap)
```

Comment:
We now have three sort-of-arrays getting passed in that map to each other. How about we create a new input type with three properties, url, path, and overrideArch and pass in a list of those?

Optionally it could also have additional properties for Installer and DetectedArch and this method wouldn't need any out parameters, those things could just be set on the input objects themselves